fetcher/bin/fetch_gha_min_cost.py

```diff
@@ -0,0 +1,125 @@
+import os
+import json
+from datetime import datetime, timezone, timedelta
+import concurrent.futures
+from fetcher.services.github_service import GithubService
+import logging
+
+
+logging.basicConfig(
+    level=logging.INFO, format="%(asctime)s - %(name)s - %(levelname)s - %(message)s"
+)
+logger = logging.getLogger(__name__)
+
+
+def get_environment_variables() -> str:
+    github_app_moj_token = os.getenv("GH_APP_MOJ_TOKEN")
+    if not github_app_moj_token:
+        raise ValueError(
+            "The env variable GH_APP_MOJ_TOKEN is empty or missing")
+    github_app_ap_token = os.getenv("GH_APP_AS_TOKEN")
+    if not github_app_ap_token:
+        raise ValueError(
+            "The env variable GH_APP_AS_TOKEN is empty or missing")
+
+    return github_app_moj_token, github_app_ap_token
+
+
+def calculate_gha_cost_per_run(run_id, repo_object, github_service, os_multipliers, minute_cost_usd):
+
+    total_minutes = 0
+    cost_per_run = 0
+    response = github_service.get_workflow_run_details(repo_name=repo_object.full_name, run_id=run_id)
+
+    for os_type, multiplier in os_multipliers.items():
+        billable_data = response["billable"].get(os_type)
+        if billable_data:
+            total_ms = billable_data['total_ms']
+            total_minutes += total_ms / 60000
+            cost_per_run += total_minutes * minute_cost_usd * multiplier
+
+    return total_minutes, cost_per_run
+
+
+def calculcate_gha_cost_per_repo(repo_object, start_date, end_date, github_service, os_multipliers, minute_cost_usd):
```

Comment:
type hints might be useful here, presumably the start and end dates are going to be required in a certain format

Reply:
Fixed